### PR TITLE
Feature.multiple expected

### PR DESCRIPTION
**Pull request by jneen**
Two important error message improvements:

* collect *all* failures from the same point (since many languages have a common `alt`, we want to show *everything* that could have been there, not just the one that happened to be checked last)
* given the above, `.desc(...)` should hide all the expected messages from its parsing, since those are considered part of parsing the whole.

### CODE REVIEW

File: src/parsy/__init__.py [hunk 1]
```diff
@@ -28,26 +28,37 @@ def line_info(self):
             return '<out of bounds index {!r}>'.format(self.index)
 
     def __str__(self):
-        return 'expected {} at {}'.format(self.expected, self.line_info())
+        expected_list = [repr(e) for e in self.expected]
+        expected_list.sort()
```
Comment (bugaevc) [inline src/parsy/__init__.py]: `expected_list = sorted(repr(e) for e in self.expected)` or even `sorted(map(repr, self.expected))`

File: src/parsy/__init__.py [hunk 2]
```diff
@@ -28,26 +28,37 @@ def line_info(self):
...
+            return 'expected {} at {}'.format(expected_list[0], self.line_info())
+        else:
+            return 'expected one of {} at {}'.format(', '.join(expected_list), self.line_info())
 
 
 class Result(namedtuple('Result', 'status index value furthest expected')):
     @staticmethod
     def success(index, value):
-        return Result(True, index, value, -1, None)
+        return Result(True, index, value, -1, frozenset([]))
```
Comment (bugaevc) [inline src/parsy/__init__.py]: Just `frozenset()` should be enough to create an empty set

**Comment by jneen** [comment]
Agreed all round, but I've gotta stop for now, so I leave it to you all. Thanks for reviving this dead project :].

**Comment by spookylukey** [comment]
@jneen thanks so much for submitting these improvements, we can take it on from here!

**Pull request merged**